pkg/util/namespaces/validation.go

```diff
@@ -0,0 +1,186 @@
+package namespaces
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/loft-sh/vcluster/config"
+	"k8s.io/apimachinery/pkg/api/validation"
+)
+
+func ValidateNamespaceSyncConfig(c *config.Config, name, namespace string) error {
+	if !c.Sync.ToHost.Namespaces.Enabled {
+		return nil
+	}
+
+	configPathIdentifier := "config.sync.toHost.namespaces.mappings.byName"
+
+	if len(c.Sync.ToHost.Namespaces.Mappings.ByName) == 0 {
+		return fmt.Errorf("%s are empty", configPathIdentifier)
+	}
+
+	virtualNamespaceNames := make([]string, 0, len(c.Sync.ToHost.Namespaces.Mappings.ByName))
+	hostNamespaceNames := make([]string, 0, len(c.Sync.ToHost.Namespaces.Mappings.ByName))
```

Comment:
There's [set implementation](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set) in apimachinery I just recently learned about from another PR, that you could use to check for duplicates in a single pass.

Reply:
that's great! thx for this, didn't know about it! Updated to use sets 🫡